### PR TITLE
FIX: Check vbo size

### LINKED ISSUE #503: gloo should raise an issue when passing a NumPy array to an attribute with a different data type
For example, passing a `(2, 20)` array to a `vec2` attribute should raise an exception. For now, it works but the coordinates are wrong (the array is flattened before being uploaded to the GPU).

Related to #502.

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #503.

Should be ready for review/merge, but I'll wait until the CIs are happy to ping.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
@rossant ready for review/merge from my end.

**Comment by almarklein** [comment]
LGTM

File: vispy/gloo/tests/test_program.py [hunk 1]
```diff
@@ -175,6 +175,7 @@ def test_attributes(self):
         assert isinstance(program._user_variables['B'], VertexBuffer)
         
         # Set non-exisint uniforms
```
Comment (rossant) [inline vispy/gloo/tests/test_program.py]: typo

**Comment by rossant** [comment]
LGTM (just one tiny typo somewhere) but this might need a rebase

**Comment by Eric89GXL** [comment]
Oh yes it does, I'll rebase and fix the typo, thanks

**Pull request merged**